cmd/kube-apiserver/app/options/options.go

```diff
@@ -168,6 +171,13 @@ func (s *ServerRunOptions) AddFlags(fs *pflag.FlagSet) {
 		"of type NodePort, using this as the value of the port. If zero, the Kubernetes master "+
 		"service will be of type ClusterIP.")
 
+	fs.StringVar(&s.KubernetesServiceExternalName, "kubernetes-service-external-name", s.KubernetesServiceExternalName, ""+
+		"If non-empty, the Kubernetes master service (which apiserver creates/maintains) will be "+
+		"of type ExternalName, using this as the value of the externalName.")
+
+	fs.IntVar(&s.APIServerServicePort, "apiserver-service-port", s.APIServerServicePort, ""+
+		"Port of the apiserver service.")
```

Comment:
Right now, the port is always 443. kube-proxy does a rewrite from 443 to `--secure-port` when using ClusterIP or to `--kubernetes-service-node-port` when set.

I would like to avoid letting kube-proxy add a iptables rule to rewrite traffic to ExternName services. I think iptables does a dns lookup once on adding the rule. This would break the communication to the apiserver at some point as the ip of the AWS LB will change.